WIP BXMSDOC-6503: Document dashbuilder external components feature. (#87)
kaldesai/kie-docs: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-external-components-con.adoc

[id='building-custom-dashboard-widgets-external-components-con']
= External components

> In {CENTRAL}, a page can have external components. The components are disabled by default. To enable the external components, change the value of `dashbuilder.components.enable` system property to `true`.

Reviewer: ```suggestion
In {CENTRAL}, you can add external components to a page. The components are disabled by default. To enable the external components, change the value of `dashbuilder.components.enable` system property to `true`.
```

Author: Done.

The external component location is set and configured with the `dashbuilder.components.dir` system property. The default value of this system property is `/tmp/dashbuilder/components`.

> You must set the component under components directory with a parent directory, that will be the component's ID. For example, if the component ID is `mycomp` and the component directory is `/tmp/dashbuilder/components`, then the component base directory is `/tmp/dashbuilder/components/mycomp`.

Reviewer: ```suggestion
You must set the component under the components directory with a parent directory, which is used as the component ID. For example, if the component ID is `mycomp` and the component directory is `/tmp/dashbuilder/components`, then the component base directory is `/tmp/dashbuilder/components/mycomp`.
```

Author: Done.

{CENTRAL} checks the `manifest.json` file in the components directory. The `manifest.json` must contain at least one `name` text parameter.

.manifest.json file descriptions
[cols="1,1", options="header"]
|===
| Parameter
| Description

|`name`
|Name of the component displayed under *Components* section.

|`icon`
|Icon of the component displayed under *Components* section.

|`noData`
|A flag that indicates that the component does not require a data set.

|`parameters`

> |The list of parameters supported by the component uses `ComponentParameter` type. Supported parameter types are `name`, `type`, `category`, `defaultValue`, `label`, and `comboValues`.

Reviewer: ```suggestion
|The list of parameters using `ComponentParameter` type. Supported parameter types include `name`, `type`, `category`, `defaultValue`, `label`, and `comboValues`.
```

Author: Done
